src/clj/puppetlabs/services/puppet_admin/puppet_admin_core.clj

```diff
@@ -101,24 +99,40 @@
 ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
 ;;; Public
 
-(schema/defn ^:always-validate config->puppet-admin-settings :- PuppetAdminSettings
-  "Given the full Puppet Server config map, extract and return the settings for
-  the Puppet Admin web service."
+(schema/defn ^:always-validate config->puppet-admin-settings
+  :- (schema/maybe PuppetAdminSettings)
+  "Given the full Puppet Server config map, extract and return the Puppet
+  Admin web service settings."
   [config]
-  (if-let [settings (:puppet-admin config)]
-    settings
-    (throw (IllegalArgumentException.
-             (str "'puppet admin' section required but not found in "
-                  "configuration settings")))))
+  (:puppet-admin config))
+
+(schema/defn ^:always-validate puppet-admin-settings->whitelist-settings
+  :- ringutils/WhitelistSettings
+  "Given the full Puppet Admin web service settings, extract and return the
+  embedded client whitelist settings."
+  [admin-settings :- (schema/maybe PuppetAdminSettings)]
+  (select-keys admin-settings [:client-whitelist
+                               :authorization-required]))
```

Comment:
Not that we expect the keys to change, but you might get the keys from `ringutils/WhitelistSettings` rather than hardcode them again here.